src/svgs/runin/runin-shmetro.tsx

```diff
@@ -235,17 +329,153 @@ const NextText = (props: { nextName: Name } & React.SVGProps<SVGGElement>) => {
         </g>
     );
 };
+const StationNumberText = (props: { lineNum: string, stationNum: string } & React.SVGProps<SVGGElement>) => {
+    const { lineNum, stationNum, ...others } = props;
+    return (
+        <g textAnchor="middle" {...others}>
+            <text className="rmg-station-name" fontSize={22}
+                dy={0}>
+                {lineNum}
+            </text>
+            <text className="rmg-station-name" fontSize={22}
+                dy={28}>
+                {stationNum}
+            </text>
+        </g>
+    );
+};
+
+const StationNumberTextNow = (props: { lineNum: string, stationNum: string } & React.SVGProps<SVGGElement>) => {
+    const { lineNum, stationNum, ...others } = props;
+    return (
+        <g textAnchor="middle" {...others}>
+            <text className="rmg-station-name" fontSize={29}
+                dy={0}>
+                {lineNum}
+            </text>
+            <text className="rmg-station-name" fontSize={29}
+                dy={35}>
+                {stationNum}
+            </text>
+        </g>
+    );
+};
+
+
+const StationNumberNext = (props: { xpos: number, ypos: number, lineName: string, stationNumber: string } & React.SVGProps<SVGGElement>) => {
```

Comment:
The same as above. All 4 components share the vast majority and can be controlled with some switch line.